R/display.R

```diff
@@ -1,74 +1,171 @@
+namedlist <- function() setNames(list(), character(0))
+
 base_display <- function(data, metadata) {
     warning("IR_display can only be used from the IPython R kernel and R magic.")
 }
 
 #'Display data by mimetype, with optional alternative representations.
 #'
-#' @param data A named list mapping mimetypes to content (base64 encoded for binary data)
-#' @param metadata A named list mapping mimetypes to named lists of specific metadata
+#' @param data  A named list mapping mimetypes to content (base64 encoded for binary data)
+#' @param metadata  A named list mapping mimetypes to named lists of specific metadata
 #' @export
-display <- function(data, metadata=NULL) {
+display <- function(data, metadata = NULL) {
     base_display(data, metadata)
 }
 
 #'Display a single type of data
 #'
-#' @param mimetype The mimetype of the data
-#' @param  content The data as a length 1 string vector
-#' @param  metadata A named list of metadata
+#' @param mimetype  The mimetype of the data
+#' @param content  The data as a length 1 string vector
+#' @param metadata  A named list of metadata
 #' @export
-display1 <- function(mimetype, content, metadata=NULL) {
-    data = list()
-    data[mimetype] = content
+display1 <- function(mimetype, content, metadata = NULL) {
+    data <- list()
+    data[[mimetype]] <- content
     base_display(data, metadata)
 }
 
-#'Display HTML output
+get_display_fun <- function(format) switch(
+    format,
+    png = display_png,
+    pdf = display_pdf,
+    svg = display_svg,
+    stop(sprintf('invalid format “%s” specified', format)))
+
+#'Display a set of image formats that the output format can select from
 #'
-#' @param  content The HTML as a length 1 string vector
+#' @param ...  Formats and their parameters.
+#' Parameters have to be lists if more/other than the first one (data) is given.
+#' 
+#' Allowed formats: png, pdf, svg
+#' 
+#' @examples
+#' \dontrun{
+#' display_images(
+#'   pdf = list(file = 'test.pdf'),
+#'   png = list(some.pngdata, width = 800, height = 600),
+#'   svg = '<svg></svg>')
+#' }
+#' 
 #' @export
-display_html = function(content) {
-    display1('text/html', content)
+display_images <- function(...) {
```

Comment:
I'm not quite sure about this function. Can you pull this (and related changes like `get_display_fun` out into a separate PR so we can discuss them apart from the more mundane changes to formatting etc.